### PR TITLE
whistle: remove depends_on arch: :x86_64 & add install option "--production"

### PR DESCRIPTION
**Pull request by avwo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/whistle.rb [hunk 1]
```diff
@@ -3,21 +3,18 @@
...
   homepage "https://github.com/avwo/whistle"
-  url "https://registry.npmjs.org/whistle/-/whistle-2.9.22.tgz"
-  sha256 "bcaa19844acb08527091d7d6193bd027e45782011cf17cb13b564a5b258c392e"
+  url "https://github.com/avwo/whistle/archive/refs/tags/v2.9.24.tar.gz"
+  sha256 "e1d5892b300227c658ec7975c139cabf13c4c1b07abd9501a7ebedab52632790"
   license "MIT"
 
   bottle do
-    sha256 cellar: :any_skip_relocation, all: "1b8d58714e6b860fe5dac96826c69f30a82a9abcd697bfe240e3d780a234f33c"
+    sha256 cellar: :any_skip_relocation, all: "fbc6574d7d2385a318ea0c6442b3bc7d54d7d294d22dbd82aa889ccaab4c7324"
```
Comment (SMillerDev) [inline Formula/whistle.rb:11]: Please don't change the bottle block.

File: Formula/whistle.rb [hunk 2]
```diff
@@ -3,21 +3,18 @@
 class Whistle < Formula
   desc "HTTP, HTTP2, HTTPS, Websocket debugging proxy"
   homepage "https://github.com/avwo/whistle"
-  url "https://registry.npmjs.org/whistle/-/whistle-2.9.22.tgz"
-  sha256 "bcaa19844acb08527091d7d6193bd027e45782011cf17cb13b564a5b258c392e"
+  url "https://github.com/avwo/whistle/archive/refs/tags/v2.9.24.tar.gz"
+  sha256 "e1d5892b300227c658ec7975c139cabf13c4c1b07abd9501a7ebedab52632790"
```
Comment (SMillerDev) [inline Formula/whistle.rb:7]: Why not npm?

Comment (avwo) [reply Formula/whistle.rb:7]: The new version of npm package has not been released yet, first test it with git tag, and change it back to npm for the next version.

Comment (SMillerDev) [reply Formula/whistle.rb:7]: We don't ship unreleases software in Homebrew.

Comment (avwo) [reply Formula/whistle.rb:7]: OK, can I roll back to the previous version?

**Pull request closed without merging**